daemon/api.go

```diff
@@ -770,29 +769,35 @@ func sideloadSnap(c *Command, r *http.Request) Response {
 
 	// form.File is a map of arrays of *FileHeader things
 	// we just allow one (for now at least)
+	var snapBody multipart.File
 out:
 	for _, v := range form.File {
 		for i := range v {
-			body, err = v[i].Open()
+			snapBody, err = v[i].Open()
 			if err != nil {
 				return BadRequest("cannot open POST form file: %v", err)
 			}
-			defer body.Close()
+			defer snapBody.Close()
 
 			break out
 		}
 	}
 	defer form.RemoveAll()
 
+	if snapBody == nil {
+		return BadRequest("no POST form file: %v", form)
```

Comment:
Suggested wording:
`cannot find "snap" file field in provided multipart/form-data payload"